lib/utils/metrics.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+const assert = require('assert');
+const uuid = require('uuid');
+
+/** @typedef { import('bmutex').Lock } Lock */
+
+const metrics = exports;
+
+/**
+ * @typedef {Object} Honey
+ */
+
+class Event {
+  /**
+   *
+   * @param {string} name the name for the event
+   * @param {Event | Honey} parent a parent Event or a libhoney instance
+   */
+  constructor(name, parent) {
+    assert(name, 'Event name is required');
+    assert(parent, 'Parent event or libhoney instance is required.');
+    if (parent instanceof Event) {
+      this.honey = parent.honey;
+      this.parent = parent;
+      this.trace_id = parent.trace_id;
```

Comment:
This will be sick. Would be good to link to the relevant part of the honeycomb docs that explain these magic keys. Have you checked to see if the underlying honeycomb event this class wraps has helpers to set the span and trace id so we don't need to rely on these magic values?

Reply:
I didn't see any when I looked.